common/core/desktop/src/kmx/kmx_processor.cpp

```diff
@@ -319,14 +347,26 @@ KMX_BOOL KMX_Processor::ProcessGroup(LPGROUP gp, KMX_BOOL *pOutputKeystroke)
 
   /* Use PostString to post the rest of the output string. */
 
-  if(PostString(p, m_keyboard.Keyboard, NULL, pOutputKeystroke) == psrCheckMatches)
-  {
-    if(gp->dpMatch && *gp->dpMatch)
-    {
-      PostString(gp->dpMatch, m_keyboard.Keyboard, NULL, pOutputKeystroke);
+  KMX_BOOL shouldProcessNomatch = PostString(p, m_keyboard.Keyboard, NULL, pOutputKeystroke) == psrCheckMatches;
+
+  if(m_debug_items) {
+    m_debug_items->push_rule_exit(gp, kkp, m_miniContext, m_indexStack);
+  }
```

Comment:
Corresponds to:
https://github.com/keymanapp/keyman/blob/75e47c1ccc36e1698bafa8178acf8140f88e4e2b/windows/src/engine/keyman32/kmprocess.cpp#L483

Slight refactor here to reduce WETness.